### PR TITLE
upgrade nix to 0.28.0

### PR DESCRIPTION
**Pull request by selfisekai**
this fixes builds on s390x+musl

### CODE REVIEW

File: src/linux.rs [hunk 1]
```diff
@@ -73,7 +72,7 @@ pub fn linux_ifaddresses(if_name: &str) -> Result<IfAddrs, Box<dyn std::error::E
                 if let Some(inet_addr) = address.as_sockaddr_in() {
                     add_to_types_mat(
                         AF_INET,
-                        &ip_to_string(inet_addr.ip()),
+                        &inet_addr.ip(),
```
Comment (SamuelYvon) [inline src/linux.rs:75]: This breaks the address display. Please revert

**Review by SamuelYvon — CHANGES_REQUESTED** [review]

File: src/linux.rs (hunk 1 above)
Comment (selfisekai) [reply src/linux.rs:75]: the returned value is not `u32` anymore starting nix 0.28.0, it's [`core::net::Ipv4Addr`](https://doc.rust-lang.org/stable/core/net/struct.Ipv4Addr.html)

Comment (SamuelYvon) [reply src/linux.rs]: Yep, just realized.

**Comment by SamuelYvon** [comment]
Thanks for the contribution 👍🏼

**Pull request merged**